content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/AjaxContentEntityProvider.java

```diff
@@ -0,0 +1,113 @@
+package org.sakaiproject.content.entityproviders;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.content.api.ContentTypeImageService;
+import static org.sakaiproject.content.tool.FilePickerAction.STATE_GOOGLEDRIVE_CHILDREN;
+import static org.sakaiproject.content.tool.FilePickerAction.STATE_GOOGLEDRIVE_ITEMS;
+import org.sakaiproject.entitybroker.EntityReference;
+import org.sakaiproject.entitybroker.EntityView;
+import org.sakaiproject.entitybroker.entityprovider.EntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.annotations.EntityCustomAction;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.ActionsExecutable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.AutoRegisterEntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Describeable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Outputable;
+import org.sakaiproject.entitybroker.entityprovider.extension.Formats;
+import org.sakaiproject.entitybroker.util.AbstractEntityProvider;
+import org.sakaiproject.googledrive.model.GoogleDriveItem;
+import org.sakaiproject.googledrive.service.GoogleDriveService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolSession;
+import org.sakaiproject.user.api.UserDirectoryService;
+
+public class AjaxContentEntityProvider extends AbstractEntityProvider implements EntityProvider, AutoRegisterEntityProvider, ActionsExecutable, Outputable, Describeable {
+
+    public final static String ENTITY_PREFIX = "contentajax";
+
+    @Getter @Setter private ServerConfigurationService serverConfigurationService;
+    @Getter @Setter private GoogleDriveService googleDriveService;
+    @Getter @Setter private UserDirectoryService userDirectoryService;
+    @Getter @Setter private SessionManager sessionManager;
+    @Getter @Setter private ContentTypeImageService contentTypeImageService;
+
+    @Override
+    public String getEntityPrefix() {
+        return ENTITY_PREFIX;
+    }
+
+    @Override
+    public String[] getHandledOutputFormats() {
+        return new String[] { Formats.JSON };
+    }
+
+    @EntityCustomAction(action = "doNavigateGoogleDrive", viewKey = EntityView.VIEW_LIST)
+    public JSONArray handleChatData(EntityReference view, Map<String,Object> params) {
```

Comment:
I don't think this is a good name for that method.

Reply:
I have changed the name, how do you think about the new?